src/view/domconverter.js

```diff
@@ -243,22 +247,41 @@ export default class DomConverter {
 	 * @returns {Iterable.<Node>} DOM nodes.
 	 */
 	* viewChildrenToDom( viewElement, domDocument, options = {} ) {
-		const fillerPositionOffset = viewElement.getFillerOffset && viewElement.getFillerOffset();
+		const blockFillerOffset = viewElement.getFillerOffset && viewElement.getFillerOffset();
+		const inlineFillerOffset = options.inlineFiller && options.inlineFiller.parent == viewElement ? options.inlineFiller.offset : null;
 		let offset = 0;
 
 		for ( const childView of viewElement.getChildren() ) {
-			if ( fillerPositionOffset === offset ) {
+			if ( blockFillerOffset === offset ) {
 				yield this.blockFiller( domDocument );
 			}
 
-			yield this.viewToDom( childView, domDocument, options );
+			const domNode = this.viewToDom( childView, domDocument, options );
+
+			// We need to insert inline filler before the dom node.
+			if ( inlineFillerOffset === offset ) {
+				// If dom node that was about to return is a text node, just add inline filler to that text node.
+				if ( this.isText( domNode ) ) {
+					domNode.data = INLINE_FILLER + domNode.data;
```

Comment:
I see that until now DomConverter didn't create the inline filler. It's something new for it to start doing so. It did create block filler in the past, but I wonder what do you think about it, @pjasiun?